api/http/handler/endpoints/endpoint_status_inspect.go

```diff
@@ -103,25 +103,24 @@ func (handler *Handler) endpointStatusInspect(w http.ResponseWriter, r *http.Req
 		}
 	}
 
+	if endpoint.EdgeCheckinInterval == 0 {
+		settings, err := handler.DataStore.Settings().Settings()
+		if err != nil {
+			return &httperror.HandlerError{http.StatusInternalServerError, "Unable to retrieve settings from the database", err}
+		}
+
+		endpoint.EdgeCheckinInterval = settings.EdgeAgentCheckinInterval
+	}
+
```

Comment:
The logic is changed here. 

Old logic,  allow change for anything other than zero.
New logic,  only allow change if current setting equals zero.

Reply:
I reversed the logic to avoid a DB read if possible but keeping it equivalent, can you provide a breaking test case?

Here are the two test cases I can think of, and if I didn't make any mistakes following the logic, the output is equivalent:

----

### Test Case A (zero)
**Input:**
`endpoint.EdgeCheckInterval = 0`
`settings.EdgeAgentCheckinInterval = 5`

**Original Output:**
`CheckinInterval = 5`

**New Output:**
`CheckinInterval = 5`

----

### Test Case B (non-zero)
**Input:**
`endpoint.EdgeCheckInterval = 7`
`settings.EdgeAgentCheckinInterval = 5`

**Original Output:**
`CheckinInterval = 7`

**New Output:**
`CheckinInterval = 7`

----

Hope that makes senses, let me know if you spot any mistake or there's another case I didn't consider.

Reply:
Looking at this again, I think the change you've done is right.